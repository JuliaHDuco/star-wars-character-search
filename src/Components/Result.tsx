import * as React from 'react';
import { IPerson, } from '../types';
import Person from './Person';

interface IResultProps {
  content: IPerson
}

interface IStateProps {
  showAllInfo: boolean;
}

class Result extends React.Component <IResultProps, IStateProps> {
  constructor(props: IResultProps) {
    super(props);

    this.state = {
      showAllInfo: false,
    }

    this.handleClick = this.handleClick.bind(this);
  }

  public render() {
    const renderButton = (
      <li>
        <button onClick={this.handleClick}>{this.props.content.name}</button>
      </li>
    );

    return this.state.showAllInfo ? <Person onHide={this.handleClick} info={this.props.content} /> : renderButton;
  }

  private handleClick(e: React.MouseEvent) {
    e.preventDefault();

    this.setState({
      showAllInfo: !this.state.showAllInfo
    })
  };
}

export default Result;
