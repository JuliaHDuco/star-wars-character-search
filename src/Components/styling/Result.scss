li {
  list-style: none;
  display: inline-block;

  button {
    color: white;
    border: 1px solid white;
    border-radius: 2px;
    font-size: 1.2em;
    margin: 1em 1em 0;
    padding: 1em 1.2em;
    background: transparent;
  }
}
