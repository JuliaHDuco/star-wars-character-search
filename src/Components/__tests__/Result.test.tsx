import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { testPerson } from '../../utilities/tests/data/person';
import Result from '../Result';

test('should match snapshot', () => {
  const component = renderer.create(
    <Result content={testPerson} />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('should render Result showing the Person button', () => {
  const component = renderer.create(
    <Result content={testPerson} />
  );

  expect(component.root.findAllByType('button').length).toBe(1);
  expect(component.root.findAllByType('button')[0].props.children).toBe('Luke Skywalker');
});

test('should render Result showing the Person info', () => {
  const component = renderer.create(
    <Result content={testPerson} />
  );

  const button = component.root.findAllByType('button')[0];
  button.props.onClick({preventDefault: () => true});

  expect(component.toJSON()).toMatchSnapshot();
});

test('should hide all info on close', () => {
  const component = renderer.create(
    <Result content={testPerson} />
  );

  const button = component.root.findAllByType('button')[0];
  button.props.onClick({preventDefault: () => true});

  const closeButton = component.root.findAllByType('button')[0];
  closeButton.props.onClick({preventDefault: () => true});

  expect(component.root.findAllByType('button').length).toBe(1);
  expect(component.root.findAllByType('button')[0].props.children).toBe('Luke Skywalker');
});
