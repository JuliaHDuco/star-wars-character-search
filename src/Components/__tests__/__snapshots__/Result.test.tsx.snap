// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<li>
  <button
    onClick={[Function]}
  >
    Luke Skywalker
  </button>
</li>
`;

exports[`should render Result showing the Person info 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    Hide
  </button>
  <div>
    <div>
      birth_year
      : 
      19BBY
    </div>
    <div>
      eye_color
      : 
      blue
    </div>
    <div>
      films
      : 
      https://swapi.co/api/films/2/
      https://swapi.co/api/films/6/
      https://swapi.co/api/films/3/
      https://swapi.co/api/films/1/
      https://swapi.co/api/films/7/
    </div>
    <div>
      gender
      : 
      male
    </div>
    <div>
      hair_color
      : 
      blond
    </div>
    <div>
      height
      : 
      172
    </div>
    <div>
      homeworld
      : 
      https://swapi.co/api/planets/1/
    </div>
    <div>
      mass
      : 
      77
    </div>
    <div>
      name
      : 
      Luke Skywalker
    </div>
    <div>
      skin_color
      : 
      fair
    </div>
    <div>
      species
      : 
      https://swapi.co/api/species/1/
    </div>
  </div>
</div>
`;
